components/cnc-tech/3020-xx-0300-00.stanza

```diff
@@ -0,0 +1,96 @@
+#use-added-syntax(jitx)
+defpackage ocdb/cnc-tech/_3020-xx-0300-00:
+  import core
+  import collections
+  import jitx
+  import jitx/commands
+  import ocdb/defaults
+  import ocdb/design-vars
+  import ocdb/land-patterns
+  import ocdb/tolerance
+  import ocdb/box-symbol
+  import ocdb/rules
+  import ocdb/generic-components
+  import ocdb/symbols
+  import ocdb/land-protrusions
+
+defstruct CompAttributes :
+  a-dim:Double
+  b-dim:Double
+  c-dim:Double
```

Comment:
Do you want to ensure the value is `>= 0.0`?

Reply:
```
defstruct CompAttributes :
   a-dim:Double with: (ensure => non-negative!)
   b-dim:Double with: (ensure => non-negative!)
   c-dim:Double with: (ensure => non-negative!)
   pn:String
   desc:String
```
with 
```
defn non-negative! (description:String, v:Double) :
  if v < 0.0 :
    fatal("The value (%_) given to %_ cannot be negative." % [v, description])
```

Reply:
I'm not sure adding data checks is necessary here since it's code that's encapsulated in this file and is invalid for any value not encoded here.